src/object/mod.rs

```diff
@@ -306,6 +307,19 @@ mod traits {
         fn root<'a, C: Context<'a>>(&self, cx: &mut C) -> Root<Self> {
             Root::new(cx, self)
         }
+
+        fn call_with<'a, C, K>(&self, cx: &mut C, method: K) -> NeonResult<CallOptions<'a>>
+        where
+            C: Context<'a>,
+            K: PropertyKey,
+        {
+            let mut options = self.get::<JsFunction, _, _>(cx, method)?.call_with(cx);
+            options.this(Handle::new_internal(Self::from_raw(
+                cx.env(),
+                self.to_raw(),
+            )));
```

Comment:
`Self::from_raw` might perform extra steps--for example, unpacking a pointer on `JsBox`--that are unnecessary. Upcasting to a `JsValue` should be sufficient for binding `this`.

```suggestion
            options.this(JsValue::new_internal(self.to_raw()));
```